### DISCUSSION TITLE
Why rerank model is deprecated in knowledge base

### DISCUSSION THREAD

**Discussion by AkisAya** [langgenius/dify#10535] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

![rerank deprecated](https://github.com/user-attachments/assets/9694bde9-1c2d-4ecc-a949-59940f0042cd)

**Comment by AkisAya** [comment]
i figure out that this is a bug.
i think the deprecated hint heres tends to say the the workspace has a default rereak model, but after doing some front-end logic we find current rerank model is empty or undefined, so this current model is deprecated.
![Snipaste_2024-11-11_19-27-58](https://github.com/user-attachments/assets/5091d802-c09b-423d-a2d5-d9a8f28b9550)
but here defaultModel is actually a model struct with all fileds empty string like this
```json
{
 "provider":"",
 "model": ""
}
```
in this situation, the default model is actaully emtpy and not valid.
so here instead of showing the deprecated sign, it's more likely that we tell users to configure at least a rerank model for the workspace like we do for the text-embedding model when it's not set

**Reply by AkisAya** [reply]
i submit a issue here
https://github.com/langgenius/dify/issues/10552

**Comment by johnnashautomation** [comment]
Hi, I had the same issue in Ollama 0.0.5 version. 
In the **Plugins** page if you delete Ollama and Re-Install it. Ollama 0.0.6 will installed (by this message written) 
Then if you open **Settings** page and open **Model Provider** section and you can click on Ollama then you can see the models will appear on there. 
So that you can check your Workspaces everything should be fine.
I hope that helps
